src/devices/Mcp9808/Mcp9808.cs

```diff
@@ -0,0 +1,206 @@
+
+namespace Iot.Device.Mcp9808
+{
+    /// <summary>
+    /// Microchip's MCP9808 I2C Temp sensor
+    /// </summary>
+    public class Mcp9808 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// MCP9808 I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x18;
+
+        #region prop
+
+        /// <summary>
+        /// MCP9808 Temperature
+        /// </summary>
+        public Temperature Temperature { get => Temperature.FromCelsius(GetTemperature()); }
+
+        private bool _disable;
+
+        /// <summary>
+        /// Disable MCP9808
+        /// </summary>
+        public bool Disabled
+        {
+            get
+            {
+                return _disable;
+            }
+            set
+            {
+                SetShutdown(value);
+                _disable = value;
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the MCP9808
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Mcp9808(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+
+            Disabled = false;
+
+            if (!Init())
+            {
+                throw new Exception("Unable to identify manufacturer/device id");
+            }
+        }
+
+        /// <summary>
+        /// Checks if the device is a MCP9808
+        /// </summary>
+        /// <returns>True if device has been correctly detected</returns>
+        private bool Init()
+        {
+            if (Read16(Register8.MCP_MANUF_ID) != 0x0054)
+            {
+                return false;
+            }
+
+            if (Read16(Register8.MCP_DEVICE_ID) != 0x0400)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Return the internal resolution register
+        /// </summary>
+        /// <returns>Resolution setting</returns>
+        public byte GetResolution()
+        {
+            return Read8(Register8.MCP_RESOLUTION);
+        }
+
+        /// <summary>
+        /// Wakes-up the device
+        /// </summary>
+        public void Wake()
+        {
+            SetShutdown(false);
+
+            // Sleep 250 ms, which is the typical temperature conversion time for the highest resolution (page 3 in datasheet, tCONV)
+            Thread.Sleep(250);
+        }
+
+        /// <summary>
+        /// Shuts down the device
+        /// </summary>
+        public void Shutdown()
+        {
+            SetShutdown(true);
+        }
+
+        /// <summary>
+        /// Read MCP9808 Temperature (℃)
+        /// </summary>
+        /// <returns>Temperature</returns>
+        private double GetTemperature()
+        {
+            ushort value = Read16(Register8.MCP_AMBIENT_TEMP);
+
+            if (value == 0xFFFF)
+            {
+                return double.NaN;
```

Comment:
Is this an error state? should we throw an exception in this case?

Reply:
It means it doesn't have a current temperature ready yet. The Arduino library (Adafruit) I copied the logic from returned NaN, seemed like a good idea, but I'm open to suggestions.

Reply:
I see, that is fine then, the problem here is that I'm not 100% sure that our Temperature Units class will handle that, and the property will try to convert this value into a Temperature assuming this was Celcius.

Reply:
Assuming our Temperature class does handle it, then perhaps all we need to do here is document the property to say that if you get a nan value back (or whatever FromCelcius(double.NAN) will return) then it means that the current temperature is not ready yet.

Reply:
I can do either way, document or throw exception, whatever you think is best :)

Reply:
I think documenting is fine, let's just make sure that Temperature.FromCelcius will work when we send it NaN

Reply:
I have added a comment to this, plus I tested the logic in Temperature with NaN, it works fine, it will return NaN even after * and +.